text/0039-service-overhaul.md

```diff
@@ -0,0 +1,462 @@
+***This RFC is dependent on [RFC 45](https://github.com/tytan652/rfcs/blob/protocol_api/text/0045-protocols-in-outputs-api.md).***
+
+# Summary
+
+- Make OBS able to accept third-party service plugins
+- Register services with a unique id rather than a common one
+- A service can be provided with multiple protocols
+- Separate Twitch, Restream and YouTube integration and make them plugins
+
+# Motivation
+
+Actually even if OBS has the service API, developer can't create third-party service plugin because there is no mechanism to list those and show their own property view.
+
+Before in OBS, in the Stream settings page, there was a combobox with two choice:
+- Streaming Services
+- Custom Streaming Server
+
+Each one had his own property view because those two are the two actual registered official services.
+This behavior could enable third-party service plugin to exist.
+
+Nowadays in OBS, this page show the list of services with many new elements showed of hidden depending of the selection (like recommended settings). With also Twitch and Restream OAuth integrations. And no use of the property views provided by `rtmp-services`.
+
+We need to restore the usage of property views and the possibility to create and use service plugin.
```

Comment:
While Properties view has its uses, I don't see much of a use case for it in services - there isn't much else a service *needs* displayed. The elements listed below are better suited in a more "curated" style of controls. Though I do agree they're currently very "all over the place" and that definitely needs to be improved. The service config JSON should provide all the relevant rules as to what elements are relevant.

Reply:
Using PropertiesView is required to isolate services as plugins and enable third-party plugins.